### PR TITLE
feat: enable esm import

### PR DESCRIPTION
**Pull request by AdnoC**
https://github.com/dequelabs/axe-core-npm/issues/660

Required some funkiness to make default import work with `require`, but I think its a clean solution.

### CODE REVIEW

File: packages/puppeteer/esmTest.mjs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import defaultExport, { AxePuppeteer } from './dist/index.mjs';
+import assert from 'assert';
+
+const exportIsFunction = typeof(defaultExport) === 'function';
+const exportIsSame = defaultExport === AxePuppeteer;
+assert(exportIsFunction, 'export is not a function');
+assert(exportIsSame, 'default and named export is not the same');
```
Comment (Zidious) [inline packages/puppeteer/esmTest.mjs:7]: Looks like we're missing the `exportIsSame` assertion for the other integrations, we should probably add those too

**Review by Zidious — DISMISSED** [review]
Small things inline -

File: packages/puppeteer/esmTest.mjs (hunk 1 above)
Comment (AdnoC) [reply packages/puppeteer/esmTest.mjs:7]: I only added them when they made sense: when there is both a named and default export.

File: packages/playwright/src/index.ts [hunk 2]
```diff
@@ -10,7 +10,8 @@ import type {
   SerialSelector,
   SerialFrameSelector
 } from 'axe-core';
-import { source } from 'axe-core';
+import axe from 'axe-core';
+const { source } = axe;
```
[the remark this answers is not in the document]
Comment (AdnoC) [reply packages/playwright/src/index.ts:14]: Yes. The old `import { source } from 'axe-core'` was causing import errors when importing via esm:
```
$ node a.mjs
file:///Users/acutler/Coding/axe-core-npm/packages/playwright/dist/index.mjs:3
import { source } from "axe-core";
         ^^^^^^
SyntaxError: Named export 'source' not found. The requested module 'axe-core' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'axe-core';
const { source } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:128:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:194:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:385:24)
    at async loadESM (node:internal/process/esm_loader:88:5)
    at async handleMainPromise (node:internal/modules/run_main:61:12)
```

So, I changed it to a syntax that did work.

File: packages/playwright/tests/axe-playwright.spec.ts [hunk 3]
```diff
@@ -124,7 +124,7 @@ describe('@axe-core/playwright', () => {
 
     it('handles large results', async function () {
       /* this test handles a large amount of partial results a timeout may be required */
-      this.timeout(100_000);
+      this.timeout(1_000_000);
```
[the remark this answers is not in the document]
Comment (AdnoC) [reply packages/playwright/tests/axe-playwright.spec.ts:127]: It was failing in CI due to timeout. If you look at the currently passing CI run, it takes `84111ms` to finish.

[the remark this answers is not in the document]
Comment (AdnoC) [reply packages/playwright/tests/axe-playwright.spec.ts:127]: Note that it passed locally with the old timeout. I'm pretty sure its just CI being weird.

**Review by Zidious — APPROVED** [review]
Looking good ☕ 

Not blocking - "Closes: XXX" rather than just the ticket

**Pull request merged**